presentation-compiler/src/main/dotty/tools/pc/completions/Completions.scala

```diff
@@ -496,68 +495,64 @@ class Completions(
-              owner.info
-                .membersBasedOnFlags(
-                  Flags.ParamAccessor,
-                  Flags.EmptyFlags,
-                )
-                .headOption
-                .map(_.info)
-            owner.isClass && owner.is(Flags.Implicit) &&
-            constructorParam.exists(p =>
-              qualType.widenDealias <:< p.widenDealias
-            )
-          end isImplicitClass
-
-          def isImplicitClassMethod = sym.is(Flags.Method) && !sym.isConstructor &&
-            isImplicitClass(sym.maybeOwner)
-
-          if isExtensionMethod then
-            completionsWithSuffix(
-              sym,
-              sym.decodedName,
-              CompletionValue.Extension(_, _, _)
-            ).map(visit).forall(_ == true)
-          else if isImplicitClassMethod then
+    if completionMode.is(Mode.Scope) && query.nonEmpty then
+      val visitor = new CompilerSearchVisitor(sym =>
+        indexedContext.lookupSym(sym) match
+          case IndexedContext.Result.InScope => false
+          case _ =>
             completionsWithSuffix(
               sym,
               sym.decodedName,
-              CompletionValue.ImplicitClass(_, _, _, sym.maybeOwner),
-            ).map(visit).forall(_ == true)
-          else false,
-        )
-        Some(search.searchMethods(query, buildTargetIdentifier, visitor).nn)
-    end match
+              CompletionValue.Workspace(_, _, _, sym)
+            ).map(visit).forall(_ == true),
+      )
+      Some(search.search(query, buildTargetIdentifier, visitor).nn)
+    else if completionMode.is(Mode.Member) then
+      val visitor = new CompilerSearchVisitor(sym =>
+        def isExtensionMethod = sym.is(ExtensionMethod) &&
+          qualType.widenDealias <:< sym.extensionParam.info.widenDealias
+        def isImplicitClass(owner: Symbol) =
+          val constructorParam =
+            owner.info
+              .membersBasedOnFlags(
+                Flags.ParamAccessor,
+                Flags.EmptyFlags,
+              )
+              .headOption
+              .map(_.info)
+          owner.isClass && owner.is(Flags.Implicit) &&
+          constructorParam.exists(p =>
+            qualType.widenDealias <:< p.widenDealias
+          )
+        end isImplicitClass
+
+        def isImplicitClassMethod = sym.is(Flags.Method) && !sym.isConstructor &&
+          isImplicitClass(sym.maybeOwner)
+
+        if isExtensionMethod then
+          completionsWithSuffix(
+            sym,
+            sym.decodedName,
+            CompletionValue.Extension(_, _, _)
+          ).map(visit).forall(_ == true)
+        else if isImplicitClassMethod then
+          completionsWithSuffix(
+            sym,
+            sym.decodedName,
+            CompletionValue.ImplicitClass(_, _, _, sym.maybeOwner),
+          ).map(visit).forall(_ == true)
+        else false,
+      )
+      Some(search.searchMethods(query, buildTargetIdentifier, visitor).nn)
+    else
```

Comment:
```suggestion
    else if completionMode.is(Mode.None) then
```
This case should be used when we can't figure out the path (eg. unclosed braces)

Reply:
This actually doesn't check whether Mode is None, because completion mode is modelled as bits so if you have CompletionMode.Type.is(Mode.None) it will return true. Lets leave this as it is for this PR. I have another one which will change when we search the scope.